
<div style="background-color: #f0f0f0; padding: 10px;">


# Individual Project, DSCI 100 (80 points)

## TO GET STARTED

- Download this Jupyter Notebook to your local computer. All assignment instructions are provided within the Notebook.
- Log in to JupyterHub (the same platform you use for worksheets and tutorials).
- Create a new folder for this project and upload the downloaded Notebook into that folder.
- Rename the Notebook using a meaningful filename, for example: dsci-100-project-yourname.ipynb

You will work on this Notebook over the next few weeks. Make sure to save all changes before logging out of JupyterHub to avoid losing your work.

## INSTRUCTIONS

Complete the following questions individually using the provided data set. When prompted, fill out the cells with your code (in code cells) or text (in text cells) to answer the question. The cells will have "%%% EDIT THIS MARKDOWN CELL %%%" or "### EDIT THIS CODE CELL ###" prompting you to overwrite the contents of the cell with your answer.

Your report must be fully reproducible, and **run without error by itself with no other files**. 

## REFERENCES

Students are required to use methods discussed in DSCI 100 for the project report. However, there may be some stages where you wish to research on how to handle "messy" data, including outliers, missing data, etc. **If you refer to anything beyond DSCI 100, please include a link to the reference at the end of this report**. It does not need to be a "formal" reference like an academic paper or website, you can link to blog posts or websites like StackExchange if they are relevant to your analysis. 

## GENERATIVE AI USE

This analysis is based on methods learned in DSCI 100, so you should have all of the tools you need to write this code from scratch. You may use Generative AI (i.e., ChatGPT, Gemini, etc) as an "advanced search engine" (i.e., to help de-bug code).**You cannot use it to generate code or text on the report. Any use of Generative AI must be disclosed at the bottom of this project report**. 

An example of a disclosure of appropriate Gen AI use: *"I tried to fit a model using function X and got error Y. Gen AI helped me figure out that the error was due to a missing value."*

**Failure to disclose Generative AI use is a form of academic misconduct. Submissions suspected of inappropriately using Gen AI will be flagged by the teaching team and handled on a case-by-case basis.**


## DUE DATE
- **Final Submission: March 28th, 11:59pm**

*All dates and times listed are local time in Vancouver, BC (PT)*
</div>

<div style="background-color: #f0f0f0; padding: 10px;">
    
## Part 0: Set Up (1 point)



- Fill in your name in the cell below.


**First and Last Name**: Leo Shiraishi

<div style="background-color: #f0f0f0; padding: 10px;">

## Part 1: The Data (2 points)

We will be exploring a data set on Women's March Madness (Basketball). This adapted data set contains historical records of every NCAA Division I Women’s Basketball Tournament appearance since the tournament began in 1982 up until 2018. Sourced from official NCAA records, the adapted data set captures tournament results across more than four decades of collegiate women’s basketball. All data is sourced from the NCAA and contains the data behind the story "The Rise and Fall Of Women’s NCAA Tournament Dynasties."

The data set can be read in from the following url: https://raw.githubusercontent.com/diverse-data-hub/diversedata/refs/heads/main/data-clean/womensmarchmadness.csv

To help you with the analysis, here is a table of descriptions of each of the variables present in the model. 

| Variable           | Description                                                                                                                                                        |
|--------------------|--------------------------------------------------------------------------------------------------------------|
| year               | Year                                                                                                                                                               |
| school             | School Name                                                                                                                                                        |
| seed               | Seed. The 0 seeding designation   in 1983 notes the eight teams that played an opening-round game to become the   No. 8 seed in each region.                       |
| conference         | Conference                                                                                                                                                         |
| conf_wins          | Conference Wins                                                                                                                                                    |
| conf_losses        | Conference Losses                                                                                                                                                  |
| conf_wins_pct      | Conference Win Percentage                                                                                                                                          |
| conf_rank          | Place in Conference                                                                                                                                                |
| division           | Conference Division                                                                                                                                                |
| reg_wins           | Regional Wins                                                                                                                                                      |
| reg_losses         | Regional Losses                                                                                                                                                    |
| reg_wins_pct       | Regional Win Percentage                                                                                                                                            |
| bid                | Whether the school qualified   with an automatic bid (by winning its conference or conference tournament) or   an at-large bid. [‘at-large’, ‘auto’]               |
| first_game_at_home | Whether the school played its   first-round tournament games on its home court. [‘Y’, ‘N’]                                                                         |
| tourney_wins       | Tourney Wins                                                                                                                                                       |
| tourney_losses     | Tourney Losses                                                                                                                                                     |
| tourney_finish     | Ordered categories:   [‘opening_round_loss’ < ‘first_round_loss’ < ‘second_round_loss’ <   ‘top_16_loss’ < ‘top_8_loss’ < ‘top_4_loss’ < ‘top_2_loss’ <   ‘champ’] |
| total_wins         | Total Wins                                                                                                                                                         |
| total_losses       | Total Losses                                                                                                                                                       |
| total_wins_pct     | Total Win Percentage                                                                                                                                               |


</div>


<div style="background-color: #f0f0f0; padding: 10px;">

#### Task 1.1: Read in the Data (2 points)

- Use your knowledge of Python to load in the data from the url given.
- Print out the first 6 rows to view the structure of the data. 

</div>

In [105]:
import altair as alt
import numpy as np
import pandas as pd
url = "https://raw.githubusercontent.com/diverse-data-hub/diversedata/refs/heads/main/data-clean/womensmarchmadness.csv"
ball_data = pd.read_csv(url)
ball_data.head(6)

,year,school,seed,conference,conf_wins,conf_losses,conf_wins_pct,conf_rank,division,reg_wins,reg_losses,reg_wins_pct,bid,first_game_at_home,tourney_wins,tourney_losses,tourney_finish,total_wins,total_losses,total_wins_pct
0,1982,Arizona St.,4,Western Collegiate,NaN,NaN,NaN,NaN,NaN,23,6,79.3,at-large,Y,1,1,top_16_loss,24,7,77.4
1,1982,Auburn,7,Southeastern,NaN,NaN,NaN,NaN,NaN,24,4,85.7,at-large,N,0,1,first_round_loss,24,5,82.8
2,1982,Cheyney,2,Independent,NaN,NaN,NaN,NaN,NaN,24,2,92.3,at-large,Y,4,1,top_2_loss,28,3,90.3
3,1982,Clemson,5,Atlantic Coast,6.0,3.0,66.7,4.0,NaN,20,11,64.5,at-large,N,0,1,first_round_loss,20,12,62.5
4,1982,Drake,4,Missouri Valley,NaN,NaN,NaN,NaN,NaN,26,6,81.3,auto,Y,2,1,top_8_loss,28,7,80.0
5,1982,East Carolina,6,Independent,NaN,NaN,NaN,NaN,NaN,19,7,73.1,at-large,N,0,1,first_round_loss,19,8,70.4


-------

<div style="background-color: #f0f0f0; padding: 10px;">

## Part 2: Research Questions (4 points)

You will think about a predictive research question that's relevant to your data set! 

</div>

<div style="background-color: #f0f0f0; padding: 10px;">

#### Task 2.1: Write Your Research Question (2 points)

- From your initial view of the data set, and given the available data in the data sets, write a **predictive** research question that you may be able to explore. You response should be a single sentence, and phrased as a question. 

*For example, if I had data on flower species, petal length, and petal widths, I may wonder... "can petal width and length be used to predict the species of flower?"*

</div>

Can an NCAA Women's basketball team's regular season record predict how deep their playoff run will be in the March Madness Tournament?

<div style="background-color: #f0f0f0; padding: 10px;">

#### Task 2.2: Identifying Research Question Type (2 points)

Write your answer to the following question in the cell below. 
- Is your research question a *classification* or *regression*-type predictive question? Why? 

*Your answer should be 50 words or less (1 sentence)*

</div>

It is a classification question because we are observing which point they will reach in the tournament based on their regular season record. The predictive variable in question is categorical, not quantitative.

-----


<div style="background-color: #f0f0f0; padding: 10px;">
    
## Part 3: Tidying (5 points)

You will complete specific tasks for the chosen data set
</div>

<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 3.1: Tidy the data (5 points)

Produce a tidy data set that could be used to answer your research question. Please ensure you have at least one quantitative (numeric) and one categorical variable:
- Create new variables (if needed)
- Transform the data into a tidy form (if needed)
- Remove irrelevant columns (if needed)
- Comment your code throughout
- Show the first 6 rows of the tidied data.

</div>

In [106]:
#The comments will refer to the set of code below each comment.
#This will isolate relevant data (rData lol) from the whole dataset of ball_data.
rData = ball_data[["reg_wins_pct","tourney_finish"]]

#This will just remove all rows with N/A values.
rData = rData.dropna()

#To measure a great deep playoff run, I made a new variable "deep_run", which consists of only the teams that place in the top 8 or above that year.
#The variables will also be categorical and will fit into the categories "True" or "False".
rData["deep_run"] = rData["tourney_finish"].isin(["top_8_loss", "top_4_loss", "top_2_loss", "champ"])

#Showing first 6 rows of tidied data.
rData.head(6)

,reg_wins_pct,tourney_finish,deep_run
0,79.3,top_16_loss,False
1,85.7,first_round_loss,False
2,92.3,top_2_loss,True
3,64.5,first_round_loss,False
4,81.3,top_8_loss,True
5,73.1,first_round_loss,False


<div style="background-color: #f0f0f0; padding: 10px;">
    
## Part 4: Summarizing and Exploring (18 points)

In this section, you will explore the data and summarize some relationships. </div>

<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 4.1: Calculate summary statistics of numeric variables (2 points)

- Create **one** table that outputs the mean, minimum, and maximum of **all** numeric columns in your data, dropping the missing values if they exist.
- Show the outputted table.

</div>

In [107]:
summ = rData[["reg_wins_pct"]].agg(["mean", "min", "max"])
summ

,reg_wins_pct
mean,76.083604
min,41.400000
max,100.000000


<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 4.2: Determine the amount of missingness (2 points)

- Create **one** table that calculates the number of missing values for each column in the data set (both numeric and categorical).
- Show the outputted table. 
</div>

In [108]:
missing = rData.isna().sum().to_frame(name="missing count")
missing

,missing count
reg_wins_pct,0
tourney_finish,0
deep_run,0


<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 4.3: Summarize over levels of a categorical variable (2 points)

- Create **one** table that calculates mean of one numeric variable for each of the levels of another categorical variable. 
- Show the outputted table. 
</div>

In [109]:
mean_by_finish = (
    rData.groupby("tourney_finish")
         .agg(mean_reg_wins_pct=("reg_wins_pct", "mean"))
)
mean_by_finish

,mean_reg_wins_pct
tourney_finish,
champ,91.737838
first_round_loss,72.547880
opening_round_loss,73.450000
second_round_loss,74.979962
top_16_loss,79.446959
top_2_loss,87.989189
top_4_loss,87.289189
top_8_loss,83.981757


<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 4.4: Plot the distribution of two relevant variables in the raw data (4 points)

- Plot the distribution of one of the numeric variables in your tidied data set.

- Create a visualization showing the relative proportions of each category/level for one of categorical variable in your tidied data set. 

- Clearly label the variable that is being plotted in each plot.

*Notes:* 
- *Be sure the output is as clear as possible, provide an English title along with readable x and y column names*

- *You may also add more code cells to this section*

</div>

In [110]:
#Histogram of regular season win percentage (numeric variable).
plotA = alt.Chart(rData).mark_bar().encode(
    x=alt.X("reg_wins_pct", title="Regular Season Win Percentage"),
    y=alt.Y("count()", title="Count")
).properties(
    title="Distribution of Regular Season Win Percentage",
    width=500,
    height=300
)
plotA

alt.Chart(...)

In [111]:
#Bar chart of tournament finish categories (categorical variable).
plotB = alt.Chart(rData).mark_bar().encode(
    x=alt.X("tourney_finish",
         sort=[
            "opening_round_loss",
            "first_round_loss",
            "second_round_loss",
            "top_16_loss",
            "top_8_loss",
            "top_4_loss",
            "top_2_loss",
            "champ"
        ],
        title="Tournament Finish"),
    y=alt.Y("count()", title="Count")
).properties(
    title="Distribution of Tournament Finish",
    width=500,
    height=300
)
plotB

alt.Chart(...)

<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Part 4.5: Visualize Relationships (8 points)

Based on your research question:
- Create at least two visualizations that look at the relationship between two variables that are relevant to the research question.
- For each visualization, briefly describe how this is relevant to the analysis (use between 50 and 100 words).
- Be sure the plots are clear, concise, and readable. Provide English titles and axis labels. 

*You may add more markdown and code cells if you choose to provide more than two plots.*
</div>

This bar chart shows the average regular season win percentage for teams in each tournament finish category. It is relevant to my research question because it helps compare (the mean) overall regular season performance across different levels of playoff/tournement success. The chart suggests a positive correlation, as teams that advanced further in the tournament generally had higher average regular season win percentages than teams eliminated in earlier rounds.

In [112]:
mean_data = (
    rData.groupby("tourney_finish")
         .agg(mean_reg_wins_pct=("reg_wins_pct", "mean"))
         .reset_index()
)

plot_rel1 = alt.Chart(mean_data).mark_bar().encode(
    x=alt.X(
        "tourney_finish",
        sort=[
            "opening_round_loss",
            "first_round_loss",
            "second_round_loss",
            "top_16_loss",
            "top_8_loss",
            "top_4_loss",
            "top_2_loss",
            "champ"
        ],
        title="Tournament Finish"
    ),
    y=alt.Y("mean_reg_wins_pct",
            title="Average Regular Season Win Percentage")
).properties(
    title="Average Regular Season Win Percentage by Tournament Finish",
    width=500,
    height=350
)
plot_rel1

alt.Chart(...)

This scatter plot shows the distribution of regular season win percentages for teams across tournament finish categories. It is relevant because it helps show whether teams with stronger regular season records tend to appear more often in later playoff rounds. The plot also shows overlap between categories, suggesting that while higher win percentages are associated with deeper playoff runs, regular season performance alone does not perfectly predict a tournament finish.

In [113]:
plot_rel2 = alt.Chart(rData).mark_point(opacity=0.5).encode(
    x=alt.X(
        "tourney_finish",
        sort=[
            "opening_round_loss",
            "first_round_loss",
            "second_round_loss",
            "top_16_loss",
            "top_8_loss",
            "top_4_loss",
            "top_2_loss",
            "champ"
        ],
        title="Tournament Finish"
    ),
    y=alt.Y("reg_wins_pct", title="Regular Season Win Percentage"),
).properties(
    title="Regular Season Win Percentage by Tournament Finish",
    width=500,
    height=350
)
plot_rel2

alt.Chart(...)

------


<div style="background-color: #f0f0f0; padding: 10px;">
    
## Part 5: Analysis (35 points)

In this section, you will describe your data analysis and perform it. 
</div>

<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 5.1 Plan out your analysis (10 points)

Propose a method that was covered in class that would be appropriate to use to answer your research question. You only need to build one model with one set of predictors in this section. Write a paragraph that explains the choice of method, and be sure to answer the following questions: 

- What method do you plan to use?
- Why is this method appropriate?
- What are the potential limitations or weaknesses of the method selected?
- How are you going to process the data to apply the model? For example: Are you splitting the data? How? How many splits? What proportions will you use for the splits? At what stage will you split? Will there be a validation set? Will you use cross validation?
- How will you assess the performance of the model? Explain why the performance metric(s) chosen are appropriate.


*You answer should be around 250 words.*

</div>

I plan to use a K-nearest neighbours (K-nn) classification model to predict whether a team is likely to have a deep playoff run based on its regular season win percentage. 

This method is appropriate because the response variable, deep_run, is categorical (with two levels, True and False), making it a classification problem. 

One limitation of the K-nn model is that the results depend heavily on the choice of K, which can affect how accurate the model is. Also, this model only uses regular season win percentage as a predictor, which may not capture all the factors that influence tournament success. 

To process the data, I will split the dataset into 75% training set and 25% testing set. This will be done before fitting the model so that the test data remains separate from the training process. I will also stratify the split by the deep_run variable so that both classes are represented proportionally in both the training and test sets. After splitting, I will scale the predictor variable to remove biases. I will then use cross-validation on the training set to select the best value of K. 

To assess the performance of the model, I will use accuracy (measuring the proportion of correct predictions), because the goal is to correctly classify whether a team makes a deep playoff run. I will also examine the confusion matrix to understand how often the model incorrectly predicts each class.

<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 5.2 Perform the Analysis (20 points)

- Perform the proposed analysis.
- Be sure to comment your code throughout so that it is easy to follow. 

*You may add more code cells if needed.* 


</div>

In [114]:
# Import functions for splitting data, preprocessing, modeling, and tuning
from sklearn import set_config
from sklearn.model_selection import train_test_split, GridSearchCV
from sklearn.preprocessing import StandardScaler
from sklearn.compose import make_column_transformer
from sklearn.neighbors import KNeighborsClassifier
from sklearn.pipeline import make_pipeline

# Output pandas data frames when possible
set_config(transform_output="pandas")

# Set the random seed for reproducibility
np.random.seed(1)

# Split the data into training and testing sets
rData_train, rData_test = train_test_split(
    rData,
    train_size=0.75,
    stratify=rData["deep_run"]
)

# Create a preprocessor to standardize the numeric predictor
rData_preprocessor = make_column_transformer(
    (StandardScaler(), ["reg_wins_pct"])
)

# Create a KNN model and combine it with the preprocessor in a pipeline
knn = KNeighborsClassifier()
knn_pipe = make_pipeline(rData_preprocessor, knn)

# Create a grid of K values to test
parameter_grid = {
    "kneighborsclassifier__n_neighbors": range(1, 51, 2)
}

# Use cross-validation on the training set to choose the best K
knn_grid = GridSearchCV(
    estimator=knn_pipe,
    param_grid=parameter_grid,
    cv=5
)

# Fit the tuned model on the training data
knn_grid.fit(
    rData_train[["reg_wins_pct"]],
    rData_train["deep_run"]
)

# Display the best value of K
knn_grid.best_params_

{'kneighborsclassifier__n_neighbors': 41}

In [115]:
#This is to predict a deep playoff run on the test set.
rData_test["predicted"] = knn_grid.predict(rData_test[["reg_wins_pct"]])

#This is to create and calculate a test accuracy score
test_accuracy = knn_grid.score(
    rData_test[["reg_wins_pct"]],
    rData_test["deep_run"]
)
test_accuracy

0.8738049713193117

<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 5.3 Create a Visualization or Table of the Results of your Analysis (5 points)

- Create a visualization or table that is relevant to the results of your analysis (comment your code throughout so that it is easy to follow).
- Describe what this visualization or table is showing, and why it is important to the analysis (approx 150 words).
  

</div>

The method of showing my results is the confusion matrix shown below. It compares the actual values of deep_run to the predicted values in our K-nn classification model. The rows represent the true outcomes, while the columns represent the predictions of the model. The top left and bottom right cells represent the correct predictions, whereas the other two represent errors. This table is important because it provides a window that helps us see the model's performance based on accuracy. We may be able to get a simple number that shows the rate at which the model is accurate (test_accuracy = ~0.8738), but this table works as a good breakdown of the context of each error/accurate prediction. In this case, it is very clear that our model is significantly better at predicting non-deep playoff runs compared to the counterpart. It is important because we now know that false positives are much less common than false negatives, and we can discuss accordingly based on our biases that would otherwise not be found.

In [118]:
#This is to create our confusion matrix
conf_matrix = pd.crosstab(
    rData_test["deep_run"],
    rData_test["predicted"],
    rownames=["Actual"],
    colnames=["Predicted"]
)

conf_matrix

Predicted,False,True
Actual,,
False,443,6
True,60,14


-----

<div style="background-color: #f0f0f0; padding: 10px;">
    
## Part 6: Discuss your Results (10 points)

In this section, you will discuss some interesting findings from your analysis in Part 5 

</div>

<div style="background-color: #f0f0f0; padding: 10px;">
    
#### Task 6.1: Discuss your findings (10 points)

In 1 to 2 paragraphs (approx. 300 words or less): 
- Summarize what you found from the analysis in Part 5, in relation to your research question of interest.
- The data science workflow is an iterative process. Discuss if at any point you had to adjust your analysis plan based on the results. 
- Discuss whether this is what you expected, or if anything was surprising.
- Discuss the impact of the results, and what future questions this could lead to. 

</div>

My results of the analysis suggest that a team's win percentage has some correlation with making a deep playoff run, but it is not a strong indicator on its own. The K-nearest neighbours model achieved a reasonable accuracy overall, largely because it correctly classified most teams that did not make deep playoff runs (negative outcome); however, the confusion matrix revealed that the model struggled to identify teams that did make deep runs correctly. There was a ~81.08% chance (60/74) of the aforementioned false negatives, compared to the much less polarizing false positives at ~1.34% (6/449). This indicates that teams with a stronger regular season record tend to perform better in tournaments than weaker teams, but it is not sufficient to reliably predict deep tournament success. Most deep playoff teams are hard to identify based on regular season win percentages, so perhaps using that as the only independent variable is too conservative, and we must use other variables to provide more accurate predictions. 

During the analysis, I did not make major changes to my overall approach, as I followed the planned workflow using a K-nearest neighbours classification model. However, I did make small adjustments as I worked through the process. For example, I ensured that the data was properly split and stratified, and I used cross-validation to select the most appropriate value of K. Based on the results, a major adjustment of my plan to predict classification would require a severe overhaul of my project, down to the research question. In order to create a more accurate prediction of a deep playoff run, I would likely need to use a different independent variable, or even several different ones. 

The results were somewhat expected in my opinion. The results showed that a low-win-rate team should not make a deep playoff run, and a higher-win-rate team is much more likely to be successful, but not guaranteed. I was not very surprised by them because, as an avid NBA and WNBA fan, I know that many, many factors lead to a team's success in the playoff tournament, and not just a team's regular season success. The factors include the ones in the original dataset, such as the seeding and the conference, and others not mentioned, such as injuries and playoff-level effort. 

These results highlight the limitations of using a single variable and suggest that future analyses could include additional predictors to improve model performance and better understand what drives success in the NCAA women's basketball March Madness. To broaden our scope of discovery, in the future, we could ask questions such as, "Can an NCAA Women's basketball team's conference and seeding predict how deep their playoff run will be in the March Madness Tournament?"

<div style="background-color: #f0f0f0; padding: 10px;">
    
## REFERENCES

If you used any resources beyond the DSCI 100 content, please link them below. Failure to provide references is a form of academic misconduct. 

</div>

No outside resources. The only resources I used were the course textbook, course lectures, my notes, and Gen AI (which I mention each usecase below).

<div style="background-color: #f0f0f0; padding: 10px;">
    
## DISCLOSURE OF GENERATIVE AI USE

If you used Generative AI for any part of this project, disclose it below. 

An example of appropriate Generative AI disclosure: If you used Gen AI to debug the error, you can disclose: *"I tried to fit a model using X function and got error Y. Gen AI helped me figure out that the error was due to a missing value."*

Failure to disclose Generative AI use is a form of academic misconduct. 
</div>

For task 4.2, I tried to figure out how to get the number of missing variables in each column, but I couldn't figure it out myself. I chose to use Gen AI to help me with the solution.

For task 4.4, I used Gen AI to sort the categories. The original plot, without the sorting of the x-axis, was ordered alphabetically. This made it difficult to understand the count based on categories with inherent levels.

For task 4.5, I used Gen AI because I couldn't figure out how to create the mean of the regular win percentage without an error. With its help, I now understand that I must use .reset_index() to create a blank slate that Altair can work with.

For task 5.2, I used Gen AI to perform and create all of the pre-analysis processes. (I have had trouble on this, sorry.)

For task 5.3, I used Gen AI to help me name the column and row to make the table look nicer than the names being "predicted" and "deep_run".

<div style="background-color: #beebca; padding: 10px;">
    
## SUBMISSION (2 points)

Review your report and prepare it for the final draft. The following files should be prepared:
- This report saved as a .html file (File -> Download As -> HTML)
- The Jupyter Notebook (.ipynb) file. This file should run from top-to-bottom with no errors, and be completely reproducible.

**PLEASE SUBMIT THE FOLLOWING TO CANVAS BY MARCH 28th 11:59pm**

- The .html file of your report (1 point)
- The .ipynb file of your report (1 point)


</div>